addons/product/models/product.py

```diff
@@ -889,7 +889,7 @@ def _onchange_product_tmpl_id(self):
 
     def _sanitize_vals(self, vals):
         """Sanitize vals to sync product variant & template on read/write."""
-        if 'product_id' in vals and 'product_tmpl_id' not in vals:
+        if vals.get('product_id') and 'product_tmpl_id' not in vals:
```

Comment:
```suggestion
        if vals.get('product_id') and not vals.get('product_tmpl_id'):
```
Might as well prevent a possible issue in the future where we have a `{'product_tmpl_id': False}` for some reason?

Reply:
good idea actually, the idea is to have a `product_tmpl_id` whenever there's a `product_id` after all

updated it